File: pyg_parent/pyg_cart_interface/src/main/java/com/pyg/cart/service/CartService.java
package com.pyg.cart.service;

import com.pyg.pojogroup.Cart;

import java.util.List;

public interface CartService {
    /**
     * 添加商品到已经存在的购物车列表中
     *
     * @param cartList : 已经存在的购物车列表
     * @param itemId   : 要添加商品的sku 的id
     * @param num      : 要添加的数量
     * @return :  添加商品后的购物车列表
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

    /**
     * 从redis中查询购物车
     */
    public List<Cart> findCartListFromRedis(String username);

    /**
     * 将购物车保存到redis
     */
    public void saveCartListToRedis(String username, List<Cart> cartList);

    /**
     * 合并购物车
     */
    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);
}

File: pyg_parent/pyg_cart_web/src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService', function ($http) {

    //读取列表数据绑定到表单中
    this.findCartList = function () {
        return $http.get('../cart/findCartList.do');
    }

    this.addGoodsToCartList = function (itemId, num) {
        return $http.get('../cart/addGoodsToCartList.do?itemId=' + itemId + '&num=' + num);
    }

    //获取地址列表
    this.findAddressList = function () {
        return $http.get('address/findListByLoginUser.do');
    }

    //保存订单
    this.submitOrder = function (order) {
        return $http.post('order/add.do', order);
    }

});
